### ISSUE TITLE
Different IPC model?

### ISSUE THREAD

**Issue by andrewkittredge** [stever/AppHostCefSharp#1]
We  are looking for a replacement for WebBrowser from ExcelDNA.  This looks cool but the idea of running an http server on all of our clients is terrifying.

Does anybody know if communicating with the CefSharp process using WCF over named pipes (will|will not) work?

I might see if I can get it working.  If I get anywhere I'll put it up.

**Comment by stever** [comment]
In the past I've used CefSharp protocol handlers to route requests without HTTP. This workaround makes it more difficult to get info between browser and the add-in, however I have found a way to get messages from the client to the server... I've been meaning to expand the example with this option. As far as I remember it would allow a bound call from JS in the browser to the host add-in. I need to check again.

**Comment by andrewkittredge** [comment]
I got it working with named pipes.  I forked your project and pushed my proof of concept.  It's a disaster, it still has all the Nancy stuff, but it works on my machine.

Thank you for putting this up.

**Comment by stever** [comment]
I've pushed some changes showing strings being passed between processes. I haven't thought it out very far but I believe this could be useful. I was thinking about JSON-formatted messages as these can be easy to serialise and de-serialise to objects. Another way to go perhaps. Also likely to be a bit messy.

**Comment by andrewkittredge** [comment]
Our ExcelDNA/CefSharp integration is in production.  I wrote a re-cap @ https://groups.google.com/forum/#!topic/exceldna/PflxmqgvMkw.  Thanks again.

**Comment by stever** [comment]
Thanks for that. I've followed up on your modification and removed the HTTP server to provide a working example based on your approach. I will do some more work to remove the polling that I had adopted for shutting down the browser I think, and I'd like it to be easier to integrate with extending the browser class perhaps. However, it still serves as a good example. I wonder if administrator rights are required for the named-pipe, as that is true for the HTTP port reservation, which was not ideal anyway.
